projects/adapter/src/dbt/adapters/fal_experimental/adapter.py

```diff
@@ -80,136 +76,65 @@ def run_in_environment_with_adapter(
-    # abstractions. So this where we tranform `isolate` objects into `koldstart`
-    # objects and run the resulting isolated function.
-
-    # One big change in `koldstart` is that it supports only one target environment
-    # whereas isolate lets you stack environments on top of each other. We resolve this
-    # in two ways, for `virtualenv` environments we just merge base requirements (dbt-*)
-    # with user requirements, whereas for `conda` environments we create an `env_dict`,
-    # where we add base requirements as `pip` requirements.
-
-    if environment.target_env_type == "virtualenv":
-        requirements = []
-        for env in environment.target_environments:
-            if env.get("configuration", {}).get("requirements"):
-                requirements += env["configuration"]["requirements"]
+
+    if environment.kind == "local":
+        result = execute_model()
+        return result
+
+    if environment.kind == "virtualenv":
+        requirements = environment.config.get("requirements", [])
+        requirements += deps
         isolated_function = isolated(
             kind="virtualenv",
-            host=host,
-            requirements=requirements,
-            machine_type=environment.machine_type)(execute_model)
-    elif environment.target_env_type == "conda":
+            host=environment.host,
+            requirements=requirements
+        )(execute_model)
+    elif environment.kind == "conda":
+        dependencies = environment.config.pop("packages", [])
+        dependencies.append({"pip": deps})
         env_dict = {
             "name": "dbt_fal_env",
             "channels": ["conda-forge", "defaults"],
-            "dependencies": []
+            "dependencies": dependencies
         }
-        for env in environment.target_environments:
-            if env.get("configuration", {}).get("packages"):
-                env_dict["dependencies"] += env["configuration"]["packages"]
-            if env.get("configuration", {}).get("requirements"):
-                env_dict["dependencies"].append({"pip": env["configuration"]["requirements"]})
         isolated_function = isolated(
             kind="conda",
-            host=host,
-            env_dict=env_dict,
-            machine_type=environment.machine_type)(execute_model)
+            host=environment.host,
+            env_dict=env_dict)(execute_model)
     else:
         # We should not reach this point, because environment types are validated when the
         # environment objects are created (in utils/environments.py).
-        raise Exception(f"Environment type not supported: {environment.target_env_type}")
+        raise Exception(f"Environment type not supported: {environment.kind}")
+
+    # Machine type is only applicable in KoldstartHost
+    if type(environment.host) is KoldstartHost:
+        isolated_function = isolated_function.on(machine_type=environment.machine_type)
```

Comment:
ATL: does the isolated function error out if passed this and `LocalHost`?

Reply:
`isolated` tries to pass it as a config to the underlining environment, virtualenv or coda, and those don't expect `machine_type` on init. We should work on making config parsing much smarter in koldstart.